src/sinks/util/tcp.rs

```diff
@@ -81,13 +81,11 @@ enum TcpSinkState {
     Backoff(Delay),
 }
 
-type TcpOrTlsStream = MaybeTlsStream<
-    FramedWrite<TcpStream, BytesCodec>,
-    FramedWrite<SslStream<TcpStream>, BytesCodec>,
->;
+type TcpOrTlsStream =
+    MaybeTls<FramedWrite<TcpStream, BytesCodec>, FramedWrite<SslStream<TcpStream>, BytesCodec>>;
```

Comment:
I believe we want this to look like:

`FramedWrite<MaybeTlsStream, BytseCodec>` where `MaybeTlsStream` is an enum abstracted over a Tcp or Tls stream, but `FramedWrite` shouldn't care since they expose the same `AsyncWrite/AsyncRead`

Reply:
Ok, I can see how a `MaybeTlsStream` would be a more useful generic.

Reply:
Heh, actually, we already have that abstraction. Fixed.